multinet/util.py

```diff
@@ -98,3 +101,8 @@ def data_path(file_name: str) -> str:
     file_path = os.path.join(TEST_DATA_DIR, file_name)
     print(file_path)
     return file_path
+
+
+def generate_arango_workspace_name() -> str:
+    """Generate a string that can be used as an ArangoDB workspace name."""
+    return f"{random.choice(ascii_lowercase)}-{uuid1()}"
```

Comment:
Why pick a random letter to use as a prefix? And there is also a sort of randomness in uuid1, since it encodes the time as well. I think it would be better to simply use md5 or a sha to get a name. Technically, there can be name clashes, but in practice it's just not going to happen (and if it does, we can publish a paper with the colliding inputs :smirk_cat:).

Even better would be to figure out a reversible mapping between arbitrary names and legal Arango names. This would be a guard against database corruption (imagine if, e.g., the workspace_mapping table were lost but the data tables were all still there). To do that would require a closer study of Arango's naming rules, but it could be a good goal to have in mind since data integrity is going to matter long-term.

Reply:
Here are Arango's rules for database names: https://www.arangodb.com/docs/stable/data-modeling-naming-conventions-database-names.html

Reply:
The reason to pick a random letter is that it's a requirement of arango database names (they can't start with a number), so this is a way to get around that.

The problem with using a hash function is that renaming workspaces becomes much harder. I don't know how exactly we would actually implement it, but when changing the "external" workspace name, the hash function would produce a different "internal" name, which means we'd have to copy the arango database, which is something we should avoid. I'm not sure if that addresses your concern, but please lmk if I'm missing something.

Reply:
Ah I think I understand. So the idea is to select a legal ArangoDB name once, at the time the workspace is created, and then never again change that for the lifetime of the workspace? That is, even if the workspace name changes, only the "display name" changes, while the "internal name" would stay the same as it always was?

In that case, instead of picking a random letter, let's just use `w` (for workspace).

BTW, in order to rename the internal name as well, it would be sufficient to just rename the corresponding Arango database--I don't think you'd have to actually copy anything.

Reply:
Agree with your point on using just `w` as a prefix.

Regarding renaming the internal database name, I'm pretty sure that functionality doesn't exist. You'd have to dump and restore. See [this issue](https://github.com/arangodb/arangodb/issues/9143).

Reply:
> Regarding renaming the internal database name, I'm pretty sure that functionality doesn't exist. You'd have to dump and restore. See [this issue](https://github.com/arangodb/arangodb/issues/9143).

Good lord.